File: app/src/main/java/com/scatl/uestcbbs/module/webview/view/WebViewActivity.java
package com.scatl.uestcbbs.module.webview.view;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.KeyEvent;
import android.webkit.WebView;
import android.widget.RelativeLayout;

import com.google.android.material.appbar.MaterialToolbar;
import com.just.agentweb.AgentWeb;
import com.just.agentweb.WebChromeClient;
import com.scatl.uestcbbs.R;
import com.scatl.uestcbbs.base.BaseActivity;
import com.scatl.uestcbbs.base.BasePresenter;
import com.scatl.util.ColorUtil;
import com.scatl.uestcbbs.util.Constant;
import com.scatl.uestcbbs.util.SharePrefUtil;
import com.scatl.util.TheftProofMark;

public class WebViewActivity extends BaseActivity {

    private RelativeLayout webViewContainer;
    private MaterialToolbar toolbar;
    private AgentWeb agentWeb;
    private String url;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        try {
            TheftProofMark
                    .getInstance()
                    .setTextColor(getColor(R.color.theft_proof_color))
                    .show(this, "UID:" + SharePrefUtil.getUid(this));
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    @Override
    protected void getIntent(Intent intent) {
        super.getIntent(intent);
        url = intent.getStringExtra(Constant.IntentKey.URL);
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_web_view;
    }

    @Override
    protected void findView() {
        webViewContainer = findViewById(R.id.webview_container);
        toolbar = findViewById(R.id.toolbar);
    }

    @Override
    protected void initView() {
        super.initView();
        agentWeb = AgentWeb.with(this)
                .setAgentWebParent(webViewContainer, new RelativeLayout.LayoutParams(RelativeLayout.LayoutParams.MATCH_PARENT, RelativeLayout.LayoutParams.MATCH_PARENT))
                .useDefaultIndicator(ColorUtil.getAttrColor(this, R.attr.colorPrimary))
                .setWebChromeClient(new WebChromeClient() {
                    @Override
                    public void onReceivedTitle(WebView view, String title) {
                        toolbar.setTitle(title);
                        super.onReceivedTitle(view, title);
                    }
                })
                .createAgentWeb()
                .ready()
                .go(url);
    }

    @Override
    protected BasePresenter initPresenter() {
        return null;
    }

    @Override
    public boolean onKeyDown(int keyCode, KeyEvent event) {

        if (agentWeb.handleKeyEvent(keyCode, event)) {
            return true;
        }
        return super.onKeyDown(keyCode, event);
    }


    @Override
    protected void onDestroy() {
        super.onDestroy();
        if (agentWeb != null) agentWeb.destroy();
    }

    @Override
    protected boolean setTheftProof() {
        return true;
    }
}

File: app/src/main/java/com/scatl/uestcbbs/module/webview/view/WebViewActivity.kt
package com.scatl.uestcbbs.module.webview.view

import android.content.Intent
import android.view.KeyEvent
import android.view.MenuItem
import android.webkit.WebView
import android.widget.RelativeLayout
import com.just.agentweb.AgentWeb
import com.just.agentweb.WebChromeClient
import com.scatl.uestcbbs.R
import com.scatl.uestcbbs.base.BaseVBActivity
import com.scatl.uestcbbs.databinding.ActivityWebViewBinding
import com.scatl.uestcbbs.module.webview.presenter.WebViewPresenter
import com.scatl.uestcbbs.util.CommonUtil
import com.scatl.uestcbbs.util.Constant
import com.scatl.util.ColorUtil.getAttrColor
import com.scatl.util.SystemUtil

/**
 * Created by sca_tl at 2023/6/2 13:40
 */
class WebViewActivity: BaseVBActivity<WebViewPresenter, WebViewView, ActivityWebViewBinding>(), WebViewView {

    private var url: String? = "https://www.example.com"
    private var agentWeb: AgentWeb? = null

    override fun getViewBinding() = ActivityWebViewBinding.inflate(layoutInflater)

    override fun initPresenter() = WebViewPresenter()

    override fun getIntent(intent: Intent?) {
        super.getIntent(intent)
        intent?.let {
            url = it.getStringExtra(Constant.IntentKey.URL)
        }
    }

    override fun initView(theftProof: Boolean) {
        super.initView(false)
        agentWeb = AgentWeb
                .with(this)
                .setAgentWebParent(mBinding.container, RelativeLayout.LayoutParams(RelativeLayout.LayoutParams.MATCH_PARENT, RelativeLayout.LayoutParams.MATCH_PARENT))
                .useDefaultIndicator(getAttrColor(this, R.attr.colorPrimary))
                .setWebChromeClient(object : WebChromeClient() {
                    override fun onReceivedTitle(view: WebView, title: String) {
                        mBinding.toolbar.title = title
                        super.onReceivedTitle(view, title)
                    }
                })
                .createAgentWeb()
                .ready()
                .go(url)
    }

    override fun onOptionsSelected(item: MenuItem?) {
        if (item?.itemId == R.id.menu_open_by_system) {
            CommonUtil.openBrowser(this, url)
        }
    }

    override fun onKeyDown(keyCode: Int, event: KeyEvent?): Boolean {
        return if (agentWeb?.handleKeyEvent(keyCode, event) == true) {
            true
        } else {
            super.onKeyDown(keyCode, event)
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        agentWeb?.destroy()
    }

    override fun getContext() = this
}